packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java

```diff
@@ -553,16 +559,36 @@ public void checkResponseSchemaForGetFavorites() throws Exception
                 .field("title").is(thirdSiteModel.getTitle());
     }
 
-    @Test(groups = { TestGroup.REST_API, TestGroup.FAVORITES, TestGroup.REGRESSION })
-    @TestRail(section = { TestGroup.REST_API, TestGroup.FAVORITES }, executionType = ExecutionType.REGRESSION, description = "Verify if get favorites response returns allowableOperations object when requested")
-    public void checkResponseForGetFavoritesWithAllowableOperations()
+    @Test(groups = {TestGroup.REST_API, TestGroup.FAVORITES, TestGroup.REGRESSION})
+    @TestRail(section = {TestGroup.REST_API, TestGroup.FAVORITES}, executionType = ExecutionType.REGRESSION,
+            description = "Verify if get favorites response returns allowableOperations object when requested")
+    public void checkResponsesForGetFavoritesWithAllowableOperationsAndSearchRequestWithIsFavorite()
     {
-        RestPersonFavoritesModelsCollection adminFavorites =
+        final RestPersonFavoritesModelsCollection adminFavorites =
                 restClient.authenticateUser(adminUserModel).withCoreAPI().usingAuthUser().include(ALLOWABLE_OPERATIONS).getFavorites();
         restClient.assertStatusCodeIs(HttpStatus.OK);
 
         adminFavorites.getEntries().stream()
                 .map(RestPersonFavoritesModel::onModel)
                 .forEach(m -> m.assertThat().field(ALLOWABLE_OPERATIONS).isNotEmpty());
     }
+
+    @TestRail(section = {TestGroup.REST_API, TestGroup.FAVORITES}, executionType = ExecutionType.REGRESSION,
+            description = "Verify the get favorites request with properties parameter applied")
+    @Test(groups = {TestGroup.REST_API, TestGroup.FAVORITES, TestGroup.REGRESSION})
+    public void checkSearchResponseContainsIsFavoriteWhenRequested() throws Exception
```

Comment:
From PMD:
> +++ 1 new SignatureDeclareThrowsException violation(s) +++
> 
> packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/favorites/GetFavoritesTests.java:579 A method/constructor should not explicitly throw java.lang.Exception

Does one of the methods in this throw an exception? TBH this rule doesn't seem particularly useful for test methods.

Reply:
Oh - it turns out we already disabled it for tests, but according to the docs this only covers JUnit 3. https://docs.pmd-code.org/latest/pmd_rules_java_design.html#signaturedeclarethrowsexception

Reply:
Nevertheless, I refactored the class and removed most of `throws Exception` declarations as the were obsolete.